Generate.py

```diff
@@ -382,6 +386,28 @@ def update_weights(weights: dict, new_weights: dict, type: str, name: str) -> di
     return weights
 
 
+def roll_meta_option(option_key, game: str, category_dict: Dict) -> Any:
+    if not game:
+        return get_choice(option_key, category_dict)
+    if game in AutoWorldRegister.world_types:
+        for key, option in AutoWorldRegister.world_types[game].options.items() | Options.per_game_common_options.items():
+            if option_key == key:
```

Comment:
slow X over Y iteration when both ends support a direct lookup should be avoived